### PR TITLE
feat(notifier): add handling for emoji removal events

### PR DESCRIPTION
**Pull request by DAcodedBEAT**
## What? (description)

* Add support for handling emoji removal events.
* Introduce handleRemovedEmoji to update knownEmojis safely.
* Add logging for ignored and removed emojis.

## Why? (reasoning)

Removed emojis were missed because they were stuck in the known list even though they were removed.

### CODE REVIEW

**Comment by DAcodedBEAT** [comment]
@particledecay can you review and merge?

**Review by particledecay — APPROVED** [review]

**Comment by particledecay** [comment]
@DAcodedBEAT same here, secrets allowlisting

File: pkg/build/build.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ Built On:   {{ .Date }}`
 
 func getStringOrNotAvailable(value string) string {
 	if value != "" {
-		return fmt.Sprintf("%s", value)
+		return value
```
Comment (DAcodedBEAT) [inline pkg/build/build.go:22]: this is an issue on the main branch

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -21,6 +21,8 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```
Comment (DAcodedBEAT) [inline .github/workflows/tests.yml:25]: this is why it wasn't caught on the PR itself

**Comment by DAcodedBEAT** [comment]
@particledecay feel free to do the needful here 🫡

File: .github/workflows/tests.yml (hunk 2 above)
Comment (particledecay) [reply .github/workflows/tests.yml:25]: ah that's a good point, thanks for catching that

**Review by particledecay — APPROVED** [review]

**Pull request merged**